apps/etterlatte-saksbehandling-ui/client/src/components/behandling/revurderingsoversikt/OmgjoeringAvFarskap.tsx

```diff
@@ -0,0 +1,102 @@
+import { IDetaljertBehandling } from '~shared/types/IDetaljertBehandling'
+import { Navn, OmgjoeringAvFarskapInfo, RevurderingInfo } from '~shared/types/RevurderingInfo'
+import React, { FormEvent, useState } from 'react'
+import { BodyShort, Button, Heading } from '@navikt/ds-react'
+import { hentBehandlesFraStatus } from '~components/behandling/felles/utils'
+import { isFailure, isPending, isSuccess, useApiCall } from '~shared/hooks/useApiCall'
+import { lagreRevurderingInfo } from '~shared/api/revurdering'
+import { ApiErrorAlert } from '~ErrorBoundary'
+import { oppdaterRevurderingInfo } from '~store/reducers/BehandlingReducer'
+import styled from 'styled-components'
+import { NavnInput } from '~components/behandling/revurderingsoversikt/NavnInput'
+
+function hentUndertypeFraBehandling(behandling?: IDetaljertBehandling): OmgjoeringAvFarskapInfo | null {
```

Comment:
Hvor mange av disse funksjonene skal vi ha, en for hver type?
⛽

Reply:
Prøvde meg på ei generell løysing no, men støter på trøbbel med at interfaces i TypeScript eigentleg berre er syntaktisk sukker og at det ikkje blir halde på når det blir gjort om til JavaScript.

Hadde jo gjort seg med noko sånt, men nei.
`    function info<T extends RevurderingInfo>(key: string, type: T): T | undefined | null {
        return behandling?.revurderinginfo?.type === key ? behandling?.revurderinginfo as type : null;
    }`